Add SecureContext and API behavior of browsers

marcoscaceres opened the pull request:
Closes #25

The following tasks have been completed:

 * [x] [Modified Web platform tests](https://github.com/web-platform-tests/wpt/pull/18715)

Implementation commitment:

 * [x] WebKit (https://bugs.webkit.org/show_bug.cgi?id=201221)
 * [X] Chromium - all tests pass
 * [X] Gecko - all tests pass


<!--
    This comment and the below content is programatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/marcoscaceres/geolocation-api/pull/34.html" title="Last updated on Aug 28, 2019, 11:58 PM UTC (78a4fff)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/geolocation-api/34/eea2561...marcoscaceres:78a4fff.html" title="Last updated on Aug 28, 2019, 11:58 PM UTC (78a4fff)">Diff</a>

anssiko commented:
I think this PR would benefit from @foolip's review.

marcoscaceres commented:
Probably want a nod from @annevk too :)

annevk commented:
This doesn't end up invoking the failure callback.

marcoscaceres commented:
Will add. Totally forgot to check that 🙃

reillyeon commented:
For reference, in Blink the failure callback is called with a PositionError where code is PERMISSION_DENIED.

marcoscaceres commented:
I’ll add some tests to see what we get across browsers.

marcoscaceres commented:
Updated spec text and added tests https://github.com/web-platform-tests/wpt/pull/18715

 * Chrome and Firefox are aligned
 * Safari returns error code 2 instead of 1.

foolip commented:
idlharness.js tests will be updated by https://github.com/web-platform-tests/wpt/pull/18717 after the spec change is merged.

marcoscaceres commented:
Forgot to push before 🤪 should be good to review now.

marcoscaceres commented:
Eventually, we need to go through and rewrite all these algorithms properly.

marcoscaceres commented on index.html:
```diff
@@ -428,10 +445,25 @@ <h3><dfn>Geolocation</dfn> interface</h3>
+      </li>
+    <li>
+   <p>
+   Otherwise, immediately return a long value that uniquely identifies a <a>watch
```
Ah! oops. Fixing.

marcoscaceres commented:
@cdumez, awesome! thanks for fixing it so quickly!

marcoscaceres commented:
@reillyeon could you review the tests also? Then we should be good to land.

marcoscaceres commented:
@anssiko could you please review the tests?

anssiko commented:
@Honry PTAL

marcoscaceres commented:
Ok, tests merged. Please merge this when possible. @anssiko, could you grant me write acceptance?

anssiko commented:
Thanks @marcoscaceres for championing this fix and others for updating the implementations, we're ready to land now. @marcoscaceres you have been granted write.
